source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -646,13 +645,15 @@ static int s_increment_read_window(struct aws_channel_handler *handler, struct a
     size_t downstream_size = aws_channel_slot_downstream_read_window(slot);
     size_t current_window_size = slot->window_size;
 
-    if (downstream_size <= current_window_size) {
-        size_t likely_records_count = (downstream_size - current_window_size) % MAX_RECORD_SIZE;
+    if (downstream_size > current_window_size) {
```

Comment:
Don't you want to calculate the total desired size first, and compare against that?
```
size_t downstream_size = aws_channel_slot_downstream_read_window(slot);
size_t records_for_downstream = ceil(downstream_size / record_size);
size_t total_desired_size = downstream_size + records_for_downstream * record_size;
if (total_desired_size > current_window_size)
    ...
```